samples/azure-quantum/resource-estimation/re-with-qir.ipynb

```diff
@@ -0,0 +1,168 @@
+   "outputs": [],
+   "source": [
+    "def resource_estimation_job_from_qir(provider: AzureQuantumProvider, bitcode: bytes, **kwargs):\n",
+    "    \"\"\"A generic function to create a resource estimation job from QIR bitcode\"\"\"\n",
+    "\n",
+    "    # Find the Azure Quantum Resource Estimator target from the provider\n",
+    "    backend = provider.get_backend('microsoft.estimator')\n",
+    "\n",
+    "    # You can provide a name for the job via keyword arguments; if not,\n",
+    "    # we'll use QIR job as a default name\n",
+    "    name = kwargs.pop(\"name\", \"QIR job\")\n",
+    "\n",
+    "    # We extract some job specific arguments from the backend's configuration\n",
+    "    config = backend.configuration()\n",
+    "    blob_name = config.azure[\"blob_name\"]\n",
+    "    content_type = config.azure[\"content_type\"]\n",
+    "    provider_id = config.azure[\"provider_id\"]\n",
+    "    output_data_format = config.azure[\"output_data_format\"]\n",
+    "\n",
+    "    # Finally, we are creating the Azure Quantum JSON object and return it\n",
+    "    return AzureQuantumJob(\n",
+    "        backend=backend,\n",
+    "        target=backend.name(),\n",
+    "        name=name,\n",
+    "        input_data=bitcode,\n",
+    "        blob_name=blob_name,\n",
+    "        content_type=content_type,\n",
+    "        provider_id=provider_id,\n",
+    "        input_data_format=\"qir.v1\",\n",
+    "        output_data_format=output_data_format,\n",
+    "        input_params = kwargs,\n",
+    "        metadata={}\n",
+    "    )"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Running a sample quantum program\n",
+    "\n",
+    "Let's now create some QIR bitcode using PyQIR generator.  Here, we build a controlled S gate using 3 T gates and 2 CNOT gates."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "module = SimpleModule(\"Controlled S\", num_qubits=2, num_results=0)\n",
+    "qis = BasicQisBuilder(module.builder)\n",
+    "\n",
+    "[a, b] = module.qubits[0:2]\n",
+    "qis.t(a)\n",
+    "qis.t(b)\n",
+    "qis.cx(a, b)\n",
+    "qis.t_adj(b)\n",
+    "qis.cx(a, b)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We can now use the function we defined above together with the `bitcode` function from PyQIR to generate a resource estimation job.  We can also pass resource estimation specific arguments, e.g., setting the error rate to 1%."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "job = resource_estimation_job_from_qir(provider, module.bitcode(), errorBudget=0.01)\n",
+    "job_monitor(job)\n",
+    "result = job.result()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Finally, we print the resource estimation table."
```

Comment:
QIR is hidden from users in this example. I suggest that we have a cell that prints out generated QIR.

Reply:
The ASCII version of the bitcode is now printed in an additional cell before the job is submitted. There is also a comment that shows how to generate QIR bitcode from any QIR, which might not be generated from PyQIR.